File: my-code/1414/main.go
package main

import "fmt"

// @Time     : 2022/02/03 01:23
// @Author   : Ranshi
// @File     : main.go
// @Doc      : 1414. 和为 K 的最少斐波那契数字数目

func findMinFibonacciNumbers(k int) (ans int) {
	fibs := []int{1, 1}
	for {
		val := fibs[len(fibs)-1] + fibs[len(fibs)-2]
		if val > k {
			break
		}
		fibs = append(fibs, val)
	}
	for i := len(fibs) - 1; i >= 0; i-- {
		if k >= fibs[i] {
			ans++
			k -= fibs[i]
		}
	}
	return
}

func main() {
	fmt.Println(findMinFibonacciNumbers(19))
}

File: my-code/1414/main.py
#!/usr/bin/env python3
# -*- coding: UTF-8 -*-
# @Time     : 2021/10/31 18:00
# @Author   : Ranshi
# @File     : 123.py
# @Doc      : 1414. 和为 K 的最少斐波那契数字数目
class Solution:
    def findMinFibonacciNumbers(self, k: int) -> int:
        a = b = 1
        fibo = [a, b]
        while a + b <= k:
            fibo.append(a + b)
            a, b = b, a + b
        ans = 0
        for num in fibo[::-1]:
            if k >= num:
                ans += 1
                k -= num
        return ans


if __name__ == "__main__":
    print(Solution().findMinFibonacciNumbers(k=7))
